### PR TITLE
Adding linting checks with mypy.

### PR DESCRIPTION
**Pull request by rcolfin**
Integrating with pytest.

Updating types.

### CODE REVIEW

File: pymkv/MKVTrack.py [hunk 1]
```diff
@@ -468,17 +482,27 @@ def extract(self, output_path: str | os.PathLike | None = None, silent: bool | N
             extract_info_file += f"_{self.language}"
         if self.extension:
             extract_info_file += f".{self.extension}"
-        if (not self.language and not self.expansion) and self.track_name:
```
Comment (rcolfin) [inline pymkv/MKVTrack.py:471]: Replaced this with `self.extension`

**Comment by GitBib** [comment]
Thanks a ton for your contribution to the library! I'll definitely check out the changes you suggested soon.

File: pymkv/MKVTrack.py (hunk 1 above)
Comment (GitBib) [reply pymkv/MKVTrack.py:471]: Yeah, that's how it's supposed to be. There was a typo in the variable name.

File: pymkv/Timestamp.py [hunk 2]
```diff
@@ -371,18 +396,30 @@ def splitting_timestamp(self, timestamp: str) -> None:
 
         Example
         -------
-        >>> obj.splitting_timestamp("12:34:56.789012345")
+        >>> obj = Timestamp("12:34:56.789012345")
```
Comment (GitBib) [inline pymkv/Timestamp.py:399]: If you're gonna use >>, might as well use >> everywhere.

File: pymkv/MKVFile.py [hunk 3]
```diff
@@ -1080,10 +1161,10 @@ def order_tracks_by_file_id(self) -> None:
...
         """
-        unique_file_dict = {}
+        unique_file_dict: dict[str, int] = {}
         for track in self.tracks:
             if track.file_path not in unique_file_dict:
                 unique_file_dict[track.file_path] = len(unique_file_dict)
 
         for track in self.tracks:
-            track.file_id = unique_file_dict.get(track.file_path)
+            track.file_id = unique_file_dict[track.file_path]
```
[the remark this answers is not in the document]
Comment (rcolfin) [reply pymkv/MKVFile.py:1170]: Using get will return None but the assumption below from what I recall is that only valid values are contained in the dictionary.  I can restore the get but then it is possible to get a ValueError later on when accessed

[the remark this answers is not in the document]
Comment (GitBib) [reply pymkv/MKVFile.py:1170]: Yeah, but the error'll be right where it's supposed to be. And plenty of folks already know how to handle it.

File: pymkv/Timestamp.py (hunk 2 above)
Comment (rcolfin) [reply pymkv/Timestamp.py:399]: I enabled doctests as part of this change.  

If you use three >>> it gets interpreted as a [Pytest](https://docs.python.org/3/library/doctest.html)

There is a mix between some that are actual valid and others that are meant to be more instructive.  In this particular case I saw it as just being an example of the api.  That's what I removed one of the > so it doesn't get executed.

File: pymkv/MKVFile.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (rcolfin) [reply pymkv/MKVFile.py:1170]: How about change it to raise a ValueError in the case of a key not being found?

File: pymkv/Timestamp.py (hunk 2 above)
Comment (rcolfin) [reply pymkv/Timestamp.py:399]: [python-doctest-skip-entire-block](https://stackoverflow.com/questions/1809037/python-doctest-skip-entire-block) is quite ugly as well.

File: pymkv/MKVFile.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (rcolfin) [reply pymkv/MKVFile.py]: done

**Comment by rcolfin** [comment]
Could the Tests/run action be re-run?

**Comment by GitBib** [comment]
Just spotted a problem with the tests not passing. We need to fix it. Feel free to handle it yourself, or I'll push some fixes later that should solve the issue.

**Comment by rcolfin** [comment]
Ah, the ubuntu-latest runner tag went back from v24 to v22.
Update README.md by lkfortuna · Pull Request #10807 · actions/runner-images
<https://github.com/actions/runner-images/pull/10807>

I updated the tests.yml to instead of using ubuntu-latest to use
ubuntu-22.04 given that it has to be a stable version relative to which repo
<https://mkvtoolnix.download/downloads.html#ubuntu> mkvtoolnix is being
installed.

I pushed a change, not sure if you need to re-enable the action in the PR.

**Comment by rcolfin** [comment]
> Just spotted a problem with the tests not passing. We need to fix it. Feel free to handle it yourself, or I'll push some fixes later that should solve the issue.

The tests should pass now.  The issue is that ubuntu-latest is a moving tag for which runner it is using.  When I first started on this, the ubuntu-latest was pointing to ubuntu 24.  It appears they have rolled it back per [PR](https://github.com/actions/runner-images/pull/10807).  I therefore updated my PR to specify the ubuntu version to be 22.04 as it is in line with the [mkvtoolnix](https://mkvtoolnix.download/downloads.html#ubuntu) version used (jammy)

**Comment by GitBib** [comment]
> The tests should pass now. The issue is that ubuntu-latest is a moving tag for which runner it is using. When I first started on this, the ubuntu-latest was pointing to ubuntu 24. It appears they have rolled it back per [PR](https://github.com/actions/runner-images/pull/10807). I therefore updated my PR to specify the ubuntu version to be 22.04 as it is in line with the [mkvtoolnix](https://mkvtoolnix.download/downloads.html#ubuntu) version used (jammy)

Ubuntu version detection is now automatic, and mkvtoolnix gets installed for the right Ubuntu version on its own.

**Comment by rcolfin** [comment]
Added the **py.typed*` marker file per [PEP 0561](https://peps.python.org/pep-0561/).

**Pull request merged**